thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/AnomalyFlattenResource.java

```diff
@@ -0,0 +1,123 @@
+package org.apache.pinot.thirdeye.dashboard.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import io.swagger.annotations.ApiOperation;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.thirdeye.anomalydetection.context.AnomalyFeedback;
+import org.apache.pinot.thirdeye.common.dimension.DimensionMap;
+import org.apache.pinot.thirdeye.datalayer.bao.MergedAnomalyResultManager;
+import org.apache.pinot.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import org.apache.pinot.thirdeye.datasource.DAORegistry;
+
+
+/**
+ * Flatten the anomaly results for UI purpose.
+ * Convert a list of anomalies to rows of columns so that UI can directly convert the anomalies to table
+ */
+@Path("thirdeye/table")
+public class AnomalyFlattenResource {
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  public static final String ANOMALY_ID = "anomalyId";
+  public static final String ANOMALY_COMMENT = "comment";
+  public static final String FUNCTION_ID = "functionId";
+  public static final String WINDOW_START = "start";
+  public static final String WINDOW_END = "end";
+  public static final String DIMENSION_KEYS  = "keys";
+
+  public static final List<String> REQUIRED_JSON_KEYS =Arrays.asList(FUNCTION_ID, WINDOW_START, WINDOW_END);
+  public static final String DEFAULT_DELIMINATOR = ",";
+
+  public AnomalyFlattenResource() {
+    this.mergedAnomalyResultDAO = DAO_REGISTRY.getMergedAnomalyResultDAO();
+  }
+
+  public AnomalyFlattenResource(MergedAnomalyResultManager mergedAnomalyResultDAO) {
+    this.mergedAnomalyResultDAO = mergedAnomalyResultDAO;
+  }
+
+  /**
+   * Flatten a list of anomaly results to a list of map
+   * @param jsonPayload a json string
```

Comment:
Can you describe what these parameters hold?

Reply:
will update it. Thanks